modules/standard/Lists.chpl

```diff
@@ -0,0 +1,907 @@
+      for i in 1.._size {
+        ref item = _getRef(i);
+        _destroy(item);
+      }
+
+      //
+      // Loop through all arrays and clean them up, then clean up the "block
+      // array" itself.
+      //
+      if _arrays != nil {
+        _sanity(_totalCapacity != 0);
+        _sanity(_arrayCapacity != 0);
+        // Remember to use zero-based indexing with `_ddata`!
+        for i in 0..#_arrayCapacity {
+          ref array = _arrays[i];
+          if array == nil then
+            continue;
+          _totalCapacity -= _getArrayCapacity(i);
+          _killArray(array);
+          array = nil;
+        }
+
+        _killBlockArray(_arrays);
+        _arrays = nil;
+      }
+
+      _sanity(_totalCapacity == 0);
+      _size = 0;
+
+      _leave();
+    }
+
+    /*
+      Return a one-based index into this list of the first item whose value
+      is equal to x.
+
+      :arg x: An element to search for.
+      :arg start: The start index to start searching from.
+      :arg end: The end index to stop searching at.
+
+      :return: The index of the element to search for.
+
+      :throws IllegalArgumentError: If the given element cannot be found.
+    */
+    proc indexOf(x: eltType, start: int=1, end: int=size): int throws {
+      _enter();
+
+      try _boundsCheckLeaveOnThrow(start);
+      try _boundsCheckLeaveOnThrow(end);
+
+      for i in start..end do
+        if x == _getRef(i) {
+          _leave();
+          return i;
+        }
+
+      _leave();
+
+      const msg = "No such element in list: " + x:string;
+      throw new owned
+        IllegalArgumentError(msg);
+
+      // Should never reach here.
+      return -1;
+    }
+
+    /*
+      Return the number of times a given element is found in this list.
+
+      :arg x: An element to count.
+
+      :return: The number of times a given element is found in this list.
+      :rtype: `int`
+    */
+    proc count(x: eltType): int {
+      _enter();
+
+      var result = 0;
+
+      for i in 1.._size do
+        if x == _getRef(i) then
+          result += 1;
+
+      _leave();
+      return result;
+    }
+
+    /*
+      Sort the items of this list in place using a comparator. If no comparator
+      is provided, sort this list using the default sort order of its elements.
+
+      .. warning::
+
+        Sorting the elements of this list may invalidate existing references
+        to the elements contained in this list.
+
+      :arg comparator: A comparator used to sort this list.
+    */
+    proc sort(comparator=Sort.defaultComparator) {
+      //
+      // TODO: This is not ideal, but the Sort API needs to be adjusted before
+      // we can sort over lists directly.
+      //
+      var array = toArray();
+      clear();
+      Sort.sort(array, comparator);
+      extend(array);
+    }
+
+    /*
+      Index this list via subscript. Returns a reference to the element at a
+      given index in this list.
+
+      :arg i: The index of the element to access.
```

Comment:
"starting with 1" or something else to indicate it's 1-based?

Reply:
Isn't indexing with 1 the default?

Reply:
Yes but that doesn't mean the documentation should ignore the issue.